.github/workflows/directoriesFilesChangePR.yml

```diff
@@ -57,3 +57,53 @@ jobs:
         with:
           name: output_changed_based_build_list
           path: ./**/output_*/
+  triggerGrinderTests:
+    name: Build
+    runs-on: ubuntu-latest
+    outputs:
+      test_targets_str: ${{ steps.format_test_targets_str.outputs.test_targets_str }}
+    env:
+        TARGET_LIST:  ${{ needs.getBuildLists.outputs.buildLists }}
+    needs: getBuildLists
+    if: ${{ needs.getBuildLists.outputs.buildEnv }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: format test targets str
+      id: format_test_targets_str
+      run: |
+        python3 ./.github/workflows/createTestList.py ${{ needs.getBuildLists.outputs.buildLists }}
+    - name: trigger default job
+      uses: appleboy/jenkins-action@master
+      if: ${{contains( fromJson(needs.getBuildLists.outputs.buildLists), 'skip')}}
+      with:
+        url: "https://ci.adoptopenjdk.net/"
+        user: "xfxcwynlc"
+        token: ${{ secrets.TOKEN }}
+        job: "Grinder_Simple"
+    - name: trigger single job
+      uses: stigmelling/jenkins-action@master
+      if: ${{!contains( fromJson(needs.getBuildLists.outputs.buildLists), 'skip')}}
+      with:
+        jenkinsUrl: 'https://ci.adoptopenjdk.net/'
+        username: 'xfxcwynlc'
+        token: ${{ secrets.TOKEN }}
+        job: "Grinder_Simple"
+        params: '{"TARGET": "${{ steps.format_test_targets_str.outputs.test_targets_str}}" }'
```

Comment:
Looks like the params is something `TARGET: TESTLIST=sanity.functional,sanity.system`, which won't work. Target should be something like `TARGET:testList TESTLIST=jit_jitt,jit_recognizedMethod,testSCCMLTests2_1` https://github.com/eclipse/openj9/blob/master/test/docs/OpenJ9TestUserGuide.md#run-a-list-of-tests-
 and testlist won't work with groups. 


``

Reply:
To make the grinder job you will probably need to pass in params as a Json with "name":"value" params'. Something like {"TARGET": "sanity", "BUILD_LIST": 'openjdk'}. As we don't know if the size of `needs.getBuildLists.outputs.buildLists` it would be better to set up the job by matrix.

Reply:
I am wondering if we can just leverage the same story we use for "Rerun in Grinder" link, which is essentially a curl command with all parameters in the URL?

Short Example (with subset of params):  https://ci.adoptopenjdk.net/job/Grinder/parambuild/?JDK_VERSION=11&JDK_IMPL=hotspot&JDK_VENDOR=adoptopenjdk&BUILD_LIST=openjdk&PLATFORM=s390x_linux&TARGET=jdk_other_0


Long Example (with all params): https://ci.adoptopenjdk.net/job/Grinder/parambuild/?TARGET=extended.openjdk&SDK_RESOURCE=upstream&TEST_FLAG=&UPSTREAM_TEST_JOB_NAME=&DOCKER_REQUIRED=false&ACTIVE_NODE_TIMEOUT=0&EXTRA_DOCKER_ARGS=&VENDOR_TEST_DIRS=&TKG_OWNER_BRANCH=AdoptOpenJDK%3Amaster&PLATFORM=s390x_linux&KEEP_REPORTDIR=true&PERSONAL_BUILD=false&ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2FAdoptOpenJDK%2Fopenjdk-tests.git&LABEL=&EXTRA_OPTIONS=&CUSTOMIZED_SDK_URL=&BUILD_IDENTIFIER=&ADOPTOPENJDK_BRANCH=master&ARTIFACTORY_SERVER=&KEEP_WORKSPACE=false&JDK_VERSION=11&USER_CREDENTIALS_ID=&ITERATIONS=1&VENDOR_TEST_REPOS=&JDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptopenjdk%2Fopenjdk-jdk11u&RELEASE_TAG=&OPENJ9_BRANCH=master&OPENJ9_SHA=&JCK_GIT_REPO=&VENDOR_TEST_BRANCHES=&UPSTREAM_JOB_NAME=build-scripts%2Fjobs%2Fjdk11u%2Fjdk11u-linux-s390x-hotspot&OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse%2Fopenj9.git&CUSTOM_TARGET=&VENDOR_TEST_SHAS=&JDK_BRANCH=dev&LABEL_ADDITION=&ARTIFACTORY_REPO=&ARTIFACTORY_ROOT_DIR=&UPSTREAM_TEST_JOB_NUMBER=&DOCKERIMAGE_TAG=&JDK_IMPL=hotspot&SSH_AGENT_CREDENTIAL=&AUTO_DETECT=true&TKG_SHA=&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=&NUM_MACHINES=&OPENJDK_SHA=&BUILD_LIST=openjdk&UPSTREAM_JOB_NUMBER=822&TIME_LIMIT=10&JVM_OPTIONS=&PARALLEL=None

Reply:
@smlambert I will make the changes below and then investigate the curl command

Reply:
@sophia-guo Is there a preference towards using a curl command with parameters in the URL versus using the parameters passed as JSON?

Reply:
Does this rerun link mean we don't need `stigmelling/jenkins-action` ?

Reply:
After reading the documentation, I think a generic account is not able to fix permission issues...
* When a forked repo (like our branch) sends a pull request to the AdobeOpenJDK:master, the secret token stored in our repo are NOT passed to Action. The action running in AdobeOpenJDK:master will see an empty secret. 
<img width="472" alt="Screen Shot 2021-03-08 at 7 28 24 PM" src="https://user-images.githubusercontent.com/40300331/110399790-74c2ff80-8044-11eb-890e-e823e801c7f4.png">Reference: https://docs.github.com/en/actions/reference/encrypted-secrets

* Suppose now AdobeOpenJDK:master create the secret and in our PR we refers to the name of the secrets stored in your repo, the secret is still not passed as well. 

<img width="679" alt="Screen Shot 2021-03-08 at 7 34 50 PM" src="https://user-images.githubusercontent.com/40300331/110400271-5ad5ec80-8045-11eb-99f1-b519188a2f12.png">

By having a generic account and store the token in the repo, we can only trigger the Jenkins job when accepting and merging the PR from a fork.  At the PR stage, a Jenkins job can't be properly triggered due to the token issue. 

Related Forum Discussion:
https://github.community/t/make-secrets-available-to-builds-of-forks/16166
https://github.community/t/make-fork-secrets-available-to-fork-prs/17762
https://discuss.circleci.com/t/pull-requests-not-triggering-build/1213